mail/spring-cloud-starter-stream-source-mail/src/test/java/org/springframework/cloud/stream/app/mail/source/MailSourceConfigurationTests.java

```diff
@@ -0,0 +1,229 @@
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.pop3(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertEquals("foo\r\n", received.getPayload());
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=pop3" })
+	public static class Pop3FailTests extends MailSourceConfigurationTests {
+
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.pop3(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertTrue(!received.getPayload().equals("Test Mail"));
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=imap", "idleImap=true" })
+	public static class ImapIdlePassTests extends MailSourceConfigurationTests {
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.imap(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertEquals("foo\r\n", received.getPayload());
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=imap", "idleImap=true" })
+	public static class ImapIdleFailTests extends MailSourceConfigurationTests {
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.imap(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertTrue(!received.getPayload().equals("Test Mail"));
+		}
+
+	}
+
+	@IntegrationTest({ "protocol=imap", "java-mail-properties=" })
+	public static class BlankJavaMailPropertiesImapTests
+			extends MailSourceConfigurationTests {
+
+		@BeforeClass
+		public static void startImapServer() throws Throwable {
+			startMailServer(PoorMansMailServer.imap(0));
+		}
+
+		@Test
+		public void testSimpleTest() throws Exception {
+
+			Message<?> received = messageCollector.forChannel(source.output()).poll(10,
+					TimeUnit.SECONDS);
+			assertNotNull(received);
+			assertThat(received.getPayload(), Matchers.instanceOf(String.class));
+			assertEquals("foo\r\n", received.getPayload());
+		}
+
+	}
+
+	@SpringBootApplication
+	public static class MailSourceApplication {
+
+	}
+
+}
```

Comment:
Fixed now. Used the `eclipse-code-formatter.xml` mentioned in [https://github.com/spring-cloud/spring-cloud-starters](https://github.com/spring-cloud/spring-cloud-starters) . That had this issue.
We should fix that.